blaze/expr/core.py

```diff
@@ -411,21 +415,72 @@ def path(a, b):
     yield a
 
 
+def ordered_intersect(a, b):
+    """Set intersection of two sequences that preserves order.
+
+    Parameters
+    ----------
+    a, b : Sequence
+
+    Returns
+    -------
+    list
+
+    Examples
+    --------
+    >>> ordered_intersect('abcd', 'cdef')
+    ['c', 'd']
+    >>> ordered_intersect('bcda', 'bdfga')
+    ['b', 'd', 'a']
+    >>> ordered_intersect('zega', 'age')  # first sequence determines ordering
+    ['e', 'g', 'a']
+    """
+    common = set(a) & set(b)
+    return [x for x in unique(concat((a, b))) if x in common]
```

Comment:
could we make this lazy by default by returning a genexpr? Also maybe move to `utils` since this seems generally useful